### PR TITLE
Refactor uenclave to support multiple enclaves

### PR DESCRIPTION
**Pull request by longtomjr**
Depends on: #55 
- [x] Setup bindgen for auth-sys
 - [x] Check if the mock codegen is still needed for data-sys

### CODE REVIEW

**Comment by longtomjr** [comment]
Putting this PR in draft and changed the base since this depends on #55

**Review by Nghondzweni — APPROVED** [review]

File: rtc_uenclave/src/rtc_enclave.rs [hunk 1]
```diff
@@ -218,29 +200,15 @@ mod tests {
...
-            let create_report_ctx = ecalls::create_report_context();
-            create_report_ctx
-                .expect()
-                .return_const(Ok(EnclaveReportResult {
-                    enclave_held_data: [0; ENCLAVE_HELD_DATA_SIZE],
-                    enclave_report: sgx_report_t::default(),
-                }));
-            create_report_ctx
-        };
-
```
Comment (PiDelport) [inline rtc_uenclave/src/rtc_enclave.rs:243]: 🎉️

File: rtc_uenclave/auth-sys/build.rs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        .with_codegen_config(CodegenConfig::FUNCTIONS | CodegenConfig::TYPES)
+        .allowlist_recursively(false)
+        .array_pointers_in_arguments(true)
+        // TODO: see if there is a way to include functions using globbing
+        .allowlist_function("enclave_create_report")
+        .clang_args(&inc_args)
+        .generate()
+        .expect("Unable to generate bindings")
+        .write_to_file(out_path.join("bindings.rs"))
+        .expect("Failed to wirte bindings to file");
```
Comment (PiDelport) [inline rtc_uenclave/auth-sys/build.rs]:
```suggestion
        .expect("Failed to write bindings to file");
```

**Review by PiDelport — APPROVED** [review]
Looks good!

There's a bit of duplication between the `build.rs` files of `auth-sys` and `data-sys`, but I'm not sure if there's a sensible option to reduce this.

There's also a bit of boilerplate duplication between the `RtcAuthEnclave` and `RtcDataEnclave` implementations for the call forwarding code: we could maybe avoid this by exposing the inner RtcEnclave struct, and updating the callers to reference that directly?

**Comment by longtomjr** [comment]
> Looks good!
> 
> There's a bit of duplication between the `build.rs` files of `auth-sys` and `data-sys`, but I'm not sure if there's a sensible option to reduce this.

We should create a new crate that we can depend on as a build dependency. This is not high priority atm, but it should be a quick fix. (#64)
> 
> There's also a bit of boilerplate duplication between the `RtcAuthEnclave` and `RtcDataEnclave` implementations for the call forwarding code: we could maybe avoid this by exposing the inner RtcEnclave struct, and updating the callers to reference that directly?

There is a library that has a macro for automatically forwarding newtype methods. I prefer the duplication (since there is no duplicated functionality/logic) since it enforce us to be explicit about what methods we expose from the inner type. If this become cumbersome to deal with, we can revisit this.

**Pull request merged**